lib/tests/config_node_test.cc

```diff
@@ -158,7 +158,7 @@ void top_level_value_replace_test(shared_node_value value,
 TEST_CASE("replace nodes", "[config-node]") {
     SECTION("simple values can be replaced") {
         top_level_value_replace_test(int_node(10), int_node(15));
-        top_level_value_replace_test(long_node(numeric_limits<int>::max() + 1), int_node(1));
+        top_level_value_replace_test(long_node(int64_t(numeric_limits<int>::max()) + 1), int_node(1));
```

Comment:
Please use a static_cast here, as in `static_cast<int64_t>(numeric_limits<int>::max())`.